### DISCUSSION TITLE
Add ability to apply Hetzner labels

### DISCUSSION THREAD

**Discussion by maggie44** [kube-hetzner/terraform-hcloud-kube-hetzner#899] [Ideas]
In the Hetzner control panel there is the option of adding labels to servers. This feature is already used for some elements of the Terraform deployment, where labels are added to indicate whether a server is a control plane or node (`role=control_plane` for example). 

It would be nice to be able to add our own labels to this. The current `labels` option for the node pools and control planes only applies to the k3 and not to Hetzner. 

There is a previous request here for background:

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/375

The `extra_hetzner_labels` suggestion seems like it would make sense. 

My use case is for other Terraform and scripts that interact with the Hetzner API. It reads the current state of the services and then adds DNS entries for node pools to Cloudflare.

**Comment by mysticaltech** [comment]
Thanks @maggie44, we will do that. But about your use case, did you try external-dns (via helm), super useful and works great with cloudflare too.

**Reply by maggie44** [reply]
This looks interesting, thanks for the link. 

My first thoughts after reading through the docs is it is a little heavy for my use case, just one more thing I have to version and check for breaking changes (Cloudflare is still in beta). Each time I deploy a node pool each node gets an IP from Hetzner, I'm thinking a request to Hetzner and Cloudflare's maintained Terraform could get me up and running without all the weight and give me versioning. 

Definitely worth it being on the radar though, and going to give it some more thought, really appreciate the pointer on that one.

**Comment by supernovae** [comment]
I guess in my brain i thought if i delegated a zone from Cloudflare to Hetzner that these scripts would create DNS needed using the cluster name/domain as the prefix.  for example mycluster.domain.com being delegated from cloudflare to hetzner would have cp.mycluster.domain.com for control plane and the hostsname.mycluster.domain.com for dns names - is there any interest in such a pattern? or Is the idea to use external-dns and be done with it? :)

**Reply by mysticaltech** [reply]
@supernovae I personally like to use external-dns, because instead of breaking your back going back and forth setting dns, you just do the config one time at the ingress definition level and that's it, cloudflare gets synced automatically. Now depends if that's what you want or not.
